### PR TITLE
Add pitch support for FlxSound

### PR DESCRIPTION
**Pull request by kevinresol**
this is a hidden gem...:smirk:
reference: http://community.openfl.org/t/change-pitch-or-speed-of-audio/1118/2

### CODE REVIEW

**Comment by Gama11** [comment]
In what OpenFL version was this added (i.e., would we be dropping support for any older OpenFL version that's still working fine with flixel by adding this without any openfl version conditionals)?.

What about openfl next? In that thread Joshua seems to suggest that this is only a thing for openfl legacy.

**Comment by kevinresol** [comment]
from [this commit](https://github.com/openfl/openfl/pull/322)
we can add the legacy compiler flag maybe

File: flixel/system/FlxSound.hx [hunk 1]
```diff
@@ -629,6 +640,23 @@ class FlxSound extends FlxBasic
...
+	{
+		return _pitch;
+	}
+	
+	private function set_pitch(v:Float):Float
+	{
+		#if (cpp || neko)
+		_channel.pitch = v;
+		#else
+		FlxG.log.warn("Pitch is only supported on OpenAL targets (i.e. cpp & neko)");
```
Comment (Gama11) [inline flixel/system/FlxSound.hx]: Wondering if it wouldn't be better to only have this property exist  for cpp and neko in the first place. That's how we handle most of the input stuff in `FlxG` for example. Then again. other features like post processing use this approach (except for the warn() call), so I guess we're not really consistent about it...

File: flixel/system/FlxSound.hx [hunk 2]
```diff
@@ -79,6 +79,10 @@ class FlxSound extends FlxBasic
 	 */
 	public var volume(get, set):Float;
 	/**
+	 * Set pitch, which also alters the playback speed. Normal is 1.
+	 */
+	public var pitch(get, set):Float;
```
Comment (Gama11) [inline flixel/system/FlxSound.hx]: "Default" would probably describe this better than "Normal".

File: flixel/system/FlxSound.hx [hunk 3]
```diff
@@ -629,6 +644,20 @@ class FlxSound extends FlxBasic
...
+	
+	#if ((cpp || neko) && openfl_legacy)
+	private inline function get_pitch():Float
+	{
+		return _pitch;
+	}
+	
+	private function set_pitch(v:Float):Float
+	{
+		return _pitch = _channel.pitch = v;
```
Comment (Gama11) [inline flixel/system/FlxSound.hx]: This can produce null exceptions since `_channel` might be null.

**Comment by Gama11** [comment]
GitHub doesn't send notifications for new commits on pull requests, so it's a good idea to comment whenever you add one.

I just manually merged this: 415f410ebc9c55 Thanks for the implementation, this is a really cool feature!

**Comment by kevinresol** [comment]
@Gama11 roger that, will comment next time.

**Comment by ashes999** [comment]
Sorry to necro. @Gama11 is this in a released version of HaxeFlixel? I didn't see any mention of pitch when I looked at the api.haxeflixel.org entry for FlxSound.

**Comment by Gama11** [comment]
Yes. The api docs are currently only generated for flash.

**Comment by ashes999** [comment]
Thanks for clarifying. (We should probably have an issue to address the docs ...)

**Comment by Gama11** [comment]
We do. https://github.com/HaxeFlixel/flixel-docs/issues/132

**Pull request closed without merging**